Angular/src/app/app.component.ts

```diff
@@ -0,0 +1,111 @@
+import {NgModule, Component, enableProdMode} from '@angular/core';
+import {BrowserModule} from '@angular/platform-browser';
+import {platformBrowserDynamic} from '@angular/platform-browser-dynamic';
+import {CustomSchedulerComponent} from "./components/custom-scheduler/custom-scheduler.component";
+import {FilterFormComponent} from "./components/filter-form/filter-form.component";
+import {DxSchedulerModule, DxFormModule} from 'devextreme-angular';
+import {OptionChangedEvent} from "devextreme/ui/form";
+import {startViewDate, endViewDate} from './config';
+import {Appointment, Service, data} from "./data";
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.css'],
+  providers: [Service],
+})
+
+export class AppComponent {
+  dataSource: Appointment[];
+  currentDate: Date = new Date(2022, 9, 1);
+  useDisable: boolean = false;
```

Comment:
Let's create a special interface for the filter values (and call it FilterValues) in a separate interfaces.ts file.

We can place this file near app.component.ts:
```
app
| - components
| - app.component.css
| - app.component.html
| - app.component.ts
| - interfaces.ts <-- new interfaces.ts file
```

And in the interfaces.ts describe the new interface:
```ts
export interface FilterValues {
  query: string; // filterValue in your code
  startDate: Date;
  endDate: Date;
  useDisable: boolean;
}
```

And after it in the app.component.ts code:
```ts
import {FilterValues} from './interfaces';
// other code.
export class AppComponent {
  dataSource = data;
  currentDate: Date = new Date(2022, 9, 1);
  filterValues: FilterValues = {
    query: '',
    startDate: new Date(2022, 9, 1),
    endDate: new Date(2022, 9, 28),
    useDisable: false,
  };
  // other code.
}
```